lxd/main_init.go

```diff
@@ -428,39 +457,22 @@ they otherwise would.
 		}
 	}
 
-	if imagesAutoUpdate {
-		ss, _, err := c.GetServer()
-		if err != nil {
-			return err
-		}
-
-		if val, ok := ss.Config["images.auto_update_interval"]; ok && val == "0" {
-			err = cmd.setServerConfig(c, "images.auto_update_interval", "")
-			if err != nil {
-				return err
-			}
-		}
-	} else {
-		err = cmd.setServerConfig(c, "images.auto_update_interval", "0")
-		if err != nil {
-			return err
-		}
-	}
+	data := &cmdInitData{}
+	data.Config = map[string]interface{}{}
 
 	if networkAddress != "" {
-		err = cmd.setServerConfig(c, "core.https_address", fmt.Sprintf("%s:%d", networkAddress, networkPort))
-		if err != nil {
-			return err
-		}
-
+		data.Config["core.https_address"] = fmt.Sprintf("%s:%d", networkAddress, networkPort)
 		if trustPassword != "" {
-			err = cmd.setServerConfig(c, "core.trust_password", trustPassword)
-			if err != nil {
-				return err
-			}
+			data.Config["core.trust_password"] = trustPassword
 		}
 	}
 
+	if imagesAutoUpdate {
+		data.Config["images.auto_update_interval"] = ""
```

Comment:
Yes, it's intentional. I started assuming that we wanted this kind of behavior for *both* interactive/auto AND preseed. IOW given that the logic you mentioned has been moved further down, it will still apply to interactive/auto, but it will also apply to preseed. If my assumption is wrong and we want preseed to be more "powerful" and blindly overwrite whatever value it finds, I'll change this (and given the conversation we had a couple of days back about allow overwriting entities when possible, it's indeed probably better to change this part of the PR).